# ONE:
1) Perform sentimental analysis on the Elon-musk tweets (Exlon-musk.csv)

Importing the necessary libraries for sentiment analysis. You will need pandas, numpy, and the Natural Language Toolkit (NLTK) library.

In [1]:
import pandas as pd
import numpy as np
import nltk
nltk.download('vader_lexicon')


[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\HP\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [2]:
import chardet

# Determine the encoding of the CSV file
with open('Elon_musk.csv', 'rb') as f:
    result = chardet.detect(f.read())
    
# Read the CSV file with the detected encoding
data = pd.read_csv('Elon_musk.csv', encoding=result['encoding'])
result

{'encoding': 'Windows-1252', 'confidence': 0.7298950869484879, 'language': ''}

In [3]:
data

Unnamed: 0                                               Text
0              1                             @kunalb11 I’m an alien
1              2  @ID_AA_Carmack Ray tracing on Cyberpunk with H...
2              3                @joerogan @Spotify Great interview!
3              4                    @gtera27 Doge is underestimated
4              5  @teslacn Congratulations Tesla China for amazi...
...          ...                                                ...
1994        1995  @flcnhvy True, it sounds so surreal, but the n...
1995        1996  @PPathole Make sure to read ur terms &amp; con...
1996        1997                @TeslaGong @PPathole Samwise Gamgee
1997        1998  @PPathole Altho Dumb and Dumber is <U+0001F525...
1998        1999                          Progress update August 28

[1999 rows x 2 columns]

Preprocess the Data: Perform data preprocessing steps to clean the text data. This may include removing any irrelevant columns, removing duplicate rows, handling missing values, and transforming the text to lowercase.

In [6]:
import warnings


# Remove irrelevant columns
data = data[['Text']]

# Remove duplicate rows
data.drop_duplicates(inplace=True)

# Handle missing values
data.dropna(inplace=True)

# Convert text to lowercase
data['Text'] = data['Text'].str.lower()
data['Text']

0                                  @kunalb11 i’m an alien
1       @id_aa_carmack ray tracing on cyberpunk with h...
2                     @joerogan @spotify great interview!
3                         @gtera27 doge is underestimated
4       @teslacn congratulations tesla china for amazi...
                              ...                        
1994    @flcnhvy true, it sounds so surreal, but the n...
1995    @ppathole make sure to read ur terms &amp; con...
1996                  @teslagong @ppathole samwise gamgee
1997    @ppathole altho dumb and dumber is <u+0001f525...
1998                            progress update august 28
Name: Text, Length: 1997, dtype: object

Sentiment Analysis with NLTK: Use the NLTK library to perform sentiment analysis. 
NLTK provides a pre-trained sentiment analyzer called VADER (Valence Aware Dictionary and sEntiment Reasoner) that is suitable for this task.

In [10]:
from nltk.sentiment import SentimentIntensityAnalyzer

# Instantiate the sentiment analyzer
sid = SentimentIntensityAnalyzer()
pd.set_option('display.max_colwidth',None)
# Apply sentiment analysis to each tweet
data['sentiment_scores'] = data['Text'].apply(lambda x: sid.polarity_scores(x))
data['sentiment_scores'] 

0                 {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
1                 {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
2          {'neg': 0.0, 'neu': 0.406, 'pos': 0.594, 'compound': 0.6588}
3         {'neg': 0.412, 'neu': 0.588, 'pos': 0.0, 'compound': -0.2732}
4          {'neg': 0.0, 'neu': 0.644, 'pos': 0.356, 'compound': 0.8513}
                                     ...                               
1994    {'neg': 0.234, 'neu': 0.588, 'pos': 0.178, 'compound': -0.4215}
1995        {'neg': 0.0, 'neu': 0.658, 'pos': 0.342, 'compound': 0.636}
1996              {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
1997      {'neg': 0.537, 'neu': 0.463, 'pos': 0.0, 'compound': -0.7003}
1998       {'neg': 0.0, 'neu': 0.517, 'pos': 0.483, 'compound': 0.4215}
Name: sentiment_scores, Length: 1997, dtype: object

Extract Sentiment Labels: Extract the sentiment labels (positive, negative, or neutral) from the sentiment scores provided by the VADER analyzer.

In [11]:
# Extract the compound score from sentiment scores
data['compound_score'] = data['sentiment_scores'].apply(lambda x: x['compound'])

# Classify the sentiment labels
data['sentiment_label'] = data['compound_score'].apply(lambda x: 'positive' if x >= 0 else 'negative')


In [14]:
# Count the number of positive, negative, and neutral tweets
sentiment_counts = data['sentiment_label'].value_counts()

# Calculate the percentage of each sentiment
sentiment_percentages = sentiment_counts / sentiment_counts.sum() * 100

print("*********************************")
print("Sentiment Counts",sentiment_counts)
print("\n*********************************")
print("Sentiment Percentages",sentiment_percentages)


*********************************
Sentiment Counts positive    1766
negative     231
Name: sentiment_label, dtype: int64

*********************************
Sentiment Percentages positive    88.432649
negative    11.567351
Name: sentiment_label, dtype: float64


Analyze the Results:
As per the above sentiment labelsthe insights we gain into the sentiment expressed in Elon Musk's tweets are as follows:
out of the 1999 total tweets 1766 tweets show Positive sentiments and 231 show negative sentiments.

so 88.43% of the total tweets show Positive sentiments.

